frontend/document/document.lua

```diff
@@ -39,6 +39,8 @@ function Document:new(from_o)
     self.__index = self
     if o._init then o:_init() end
     if o.init then o:init() end
+    -- Make sure the checksum has been generated before any document saving operations.
+    o:fastDigest()
```

Comment:
Yes, that will turn it into a proper self-contained plugin, i.e. only run the code when it's enabled.